### ISSUE TITLE
Funny pattern in data from `select_mock_targets`

### ISSUE THREAD

**Issue by forero** [desihub/desitarget#414]
Running `select_mock_targets --no-spectra --nproc 4 --nside 16 --seed 10 -c ./input.yaml  --output_dir ./  --tiles ./tiles.fits` from the master version with this configuration file 

```
# mock target configuration file
targets:
    ELG: {
        target_type: ELG,
        mockfile: '{DESI_ROOT}/mocks/DarkSky/v1.0.1/elg_0_inpt.fits',
        format: gaussianfield,
        density: 2400,
    }
    LRG: {
        target_type: LRG,
        mockfile: '{DESI_ROOT}/mocks/DarkSky/v1.0.1/lrg_0_inpt.fits',
        format: gaussianfield,
        density: 480,
    }
    QSO: {
        target_type: QSO,
        mockfile: '{DESI_ROOT}/mocks/DarkSky/v1.0.1/qso_0_inpt.fits',
        format: gaussianfield,
        zmax_qso: 1.8,
        use_simqso: True,
        density: 120,
    }
    LYA: {
        target_type: QSO,
        mockfile: '{DESI_ROOT}/mocks/lya_forest/london/v4.0/master.fits',
        format: CoLoRe,
        nside_lya: 16,
        zmin_lya: 1.8,
        density: 50,
        use_simqso: True,
        balprob: 0.0,
        add_dla: True,
    }
    MWS_MAIN: {
        target_type: STAR,
        mockfile: '{DESI_ROOT}/mocks/mws/galaxia/alpha/v0.0.5/healpix',
        nside_galaxia: 8,
        format: galaxia,
    }
    WD: {
        target_type: WD,
        mockfile: '{DESI_ROOT}/mocks/mws/wd/v1.0.0/mock_wd.fits',
        format: mws_wd,
    }
    SKY: {
        target_type: SKY,
        mockfile: '{DESI_ROOT}/mocks/uniformsky/0.1/uniformsky-2048-0.1.fits',
        format: uniformsky,
    }
```

produces this output for the `targets.fits`:
![download](https://user-images.githubusercontent.com/1848618/47826489-403d2b80-dd46-11e8-97a4-4595faafa848.png)

any idea what might be happening?

**Comment by sbailey** [comment]
Does this pattern appear for all target classes, or just a single class?  Do any of the underlying mocks have this pattern?

**Comment by moustakas** [comment]
It's most prominent in the ELGs, mild in the LRGs, and not present in the QSOs, stars, or skies.  I also confirmed that it's not in the mocks themselves.

The code is parallelized over sub-healpixels, so that's what I'm going to look into next.

<img width="564" alt="screen shot 2018-11-02 at 7 13 34 pm" src="https://user-images.githubusercontent.com/1431820/47944696-6f34d800-ded3-11e8-88ed-937c8750271a.png">

**Comment by sbailey** [comment]
Fixed in PR #416
